### PR TITLE
tests: enable view tests

### PR DESCRIPTION
**Pull request by jacquerie**
It always bugged me that we have no clean way of testing authenticated views. This PR fixes this: using an idea I found on Stackoverflow, and some very helpful suggestions by @dset0x, here it is.

### CODE REVIEW

**Comment by jacquerie** [comment]
This should really be split in two commits, one fixing Travis, one adding the tests.

File: .travis.yml [hunk 1]
```diff
@@ -55,13 +55,12 @@ before_install:
...
 install:
-  - "travis_retry pip install -e git+https://github.com/inveniosoftware/invenio@master#egg=Invenio"
   - "travis_retry pip install -r .travis-$REQUIREMENTS-requirements.txt --allow-all-external"
   - "travis_retry pip install -e .[$REXTRAS]"
 
 before_script:
-  - "inveniomanage database init --yes-i-know"
-  - "inveniomanage database create --quiet"
+  - "inveniomanage database init --yes-i-know || echo ':('"
+  - "inveniomanage database create --quiet || echo ':('"
```
Comment (dset0x) [inline .travis.yml]: `false || true` will return 0 and execution will continue. Are you sure you meant to do this?

Comment (jacquerie) [inline .travis.yml]: You are right, there is no reason for this leniency.

**Comment by jacquerie** [comment]
Groan. My nifty decorator + context manager is actually useless: we just need to set `TESTING = True` in the configuration and Flask automatically disables authentication. Oh well. TIL!

**Comment by jalavik** [comment]
@jacquerie Guess this is deprecated now?

**Comment by jacquerie** [comment]
We can close this, and maybe resurrect later if needed.

**Pull request closed without merging**